hbase-client/src/main/java/org/apache/hadoop/hbase/client/Scan.java

```diff
@@ -200,35 +200,37 @@ public Scan(Scan scan) throws IOException {
     filter = scan.getFilter(); // clone?
     loadColumnFamiliesOnDemand = scan.getLoadColumnFamiliesOnDemandValue();
     consistency = scan.getConsistency();
-    this.setIsolationLevel(scan.getIsolationLevel());
     reversed = scan.isReversed();
     asyncPrefetch = scan.isAsyncPrefetch();
     small = scan.isSmall();
     allowPartialResults = scan.getAllowPartialResults();
     tr = scan.getTimeRange(); // TimeRange is immutable
-    Map<byte[], NavigableSet<byte[]>> fams = scan.getFamilyMap();
-    for (Map.Entry<byte[],NavigableSet<byte[]>> entry : fams.entrySet()) {
-      byte [] fam = entry.getKey();
-      NavigableSet<byte[]> cols = entry.getValue();
-      if (cols != null && cols.size() > 0) {
-        for (byte[] col : cols) {
-          addColumn(fam, col);
+    if (!(this instanceof ImmutableScan)) {
```

Comment:
This is not a good practise... Cna we avoid this instanceof for testing a sub class in the parent class?

Reply:
I really tried for this but didn't find better solution. For any Immutable child class, it will have to call super constructor, and here in Scan constructor, it calls setter methods, so the control will go to child class's overridden setter methods, which will fail :(